### ISSUE TITLE
Research > Systematic testing of semantic search

### ISSUE THREAD

**Issue by chrisaballard** [climatepolicyradar/policy-search#114]
Carry out some systematic tests on semantic search to identify examples of search results that work well and those that don't. This should inform the research plan and strategies to improve the search result quality.

- [x] crowdsource search queries from team with hints as to whether they worked well 
- [x] are there obvious ways we can systematically tweak or vary these?
- [x] run queries against the search index and analyse results

**Comment by kdutia** [comment]
In terms of systematically tweaking queries, I've experimented with:
- **abbreviations and acronyms** - there is no support in search for these at the moment. E.g. 'electric vehicle charging' returns completely different top results to 'EV charging'.
- **changing character case** - search queries are case-insensitive at the moment. This makes sense as something to continue with, except for maybe if we look at processing or detecting acronyms.
- **adding and removing terms from a query**
  - adding terms seems to have the desired effect of narrowing down search results, e.g. 'transport levies' -> 'transport levies in Europe'
  - when queries are very short, the search results sometimes fail to capture the intention behind both terms. E.g. 'transport levies' returns many top results about generic levies, whereas 'transport levies in Europe' returns more transport-related (as well as more Europe-related) passages. However, some of these passages aren't related to the concept of 'levies'.
- **synonyms** - the queries 'transport levies in europe' return wildly different results, with different qualities. This points to the idea that we should maybe model our key domain concepts as entities.

I've also noticed that:
- **search ranking** - often, relevant results are in the top 20 results but not near the top. This points to the fact that we should both measure and look to improve ranking.

^ I've also added and expanded these notes in [Notion](https://www.notion.so/climatepolicyradar/Systematic-testing-of-semantic-search-e5d174bd18dd4a218eba2c9f70903180)
